### ISSUE TITLE
Porting to linux

### ISSUE THREAD

**Issue by servergeeks** [nicknisi/dotfiles#30]
Hi,

Any plans on making the setup compatible with linux ?

**Comment by nicknisi** [comment]
Hi! Yes, as I've been doing more on linux these days, I will look to better automate the install process there.

**Comment by valentinul** [comment]
Cool, if you make any update please update this issue. Thank you!

**Comment by jrock2004** [comment]
One pain point is reattach-to-user-namespace. @valentinul what distro are you looking for?

**Comment by valentinul** [comment]
Mainly debian, arch would be nice also ...

**Comment by jrock2004** [comment]
Not to steal any thunder from @nicknisi but I have a similiar setup since my dotfiles were basically themed from this repo. But you can look here to see what I did https://github.com/jrock2004/dotfiles/tree/linux

**Comment by nicknisi** [comment]
I've made some updates to the install scripts to better handle `brew` not existing, and switched to zplug to load zsh-autosuggestions, nvm, and zsh-syntax-highlighting, which works on linux.

I'm still having some issues getting fzf.vim to work on linux (ubuntu). I'll take a look at this later.

**Comment by jrock2004** [comment]
On my linux side, I have to git clone fzf to the home dir then I put this in plugins.vim. I am still looking for a better solution

plugins.vim
``` 
Plug '~/.fzf/fzf' | Plug 'junegunn/fzf.vim'
```

linux.sh
```
git clone --depth 1 https://github.com/junegunn/fzf.git $HOME/.fzf
$HOME/.fzf/install
```

**Comment by arturojosejr** [comment]
Regarding git.sh you could add:
`if [ "$(uname)" == "Darwin" ]; then
  git config --global credential.helper 'osxkeychain'
  else
    read -n 1 -p "¿Guardar usuario y contraseña en un fichero sin cifrar para evitar escribirla? [s/N] "
    if [[ $REPLY =~ ^([Yy]|[Ss])$ ]]; then
      git config --global credential.helper 'store'
    else
      git config --global credential.helper 'cache --timeout=3600'
    fi
fi
`

**Comment by arturojosejr** [comment]
You have to create another tmux config file for linux ([like this](https://github.com/arturojosejr/post-install-script/blob/master/tmux/tmux.conf.symlink)) and copy it to tmux.conf.symlink if using linux. This way you correct copy/paste not working in linux.

**Comment by jrock2004** [comment]
I am currently investigating, but take a look at http://linuxbrew.sh/ . Seems like you may be able to install your apps on linux using the brew way. I am still investigating

**Comment by wilfrantz** [comment]
Hi,
Any update on this issue?

**Comment by jrock2004** [comment]
@wilfrantz what issues are you having?

**Comment by ajr-dev** [comment]
The main issue I was having was with tmux. I couldn't open tmux because of the first line in the `tmux.conf.symlink`:
```
set -g default-command "login-shell"
```
it needs to be:
```
if-shell "uname | grep -q Darwin" \
    'set -g default-command "login-shell"'
```

**Comment by wilfrantz** [comment]
@jrock2004 I mainly had to do everything manually and I also had the tmux situation. thx to @ajr-dev for the fix.

**Comment by jrock2004** [comment]
I should try setting up a VM and trying this repo again as I have customized far off from where this great repo has gone

**Comment by kinowarrior** [comment]
@jrock2004 - Hi, your link from 2016 is done (https://github.com/jrock2004/dotfiles/tree/linux). Do you still have a Linux (Ubuntu is what I'm after) fork?

**Comment by kinowarrior** [comment]
@ajr-dev - Thanks for sharing your fork - shall take a look.

**Comment by kinowarrior** [comment]
@ajr-dev - having problems  with your dotfiles for nvim with fzf

```
Error detected while processing function <SNR>42_init[6]..airline#extensions#load:
line   41:
E117: Unknown function: airline#extensions#fzf#init
```
And when i try `leader t` i see 

```
E117: Unknown function: fzf#run
```

Tried the suggestion above https://github.com/nicknisi/dotfiles/issues/30#issuecomment-274177155 but alas, no good
.

**Comment by kinowarrior** [comment]
Ok, managed to get FZF working now. The following is probably overkill, but it solved the issue:

First of all:

```
git clone --depth 1 https://github.com/junegunn/fzf.git $HOME/.fzf
$HOME/.fzf/install
```

Then ensure you have all of the following somewhere in your .vimrc ,before adding all the other fzf settings as found in this repo's dotfiles vimrc file:

```
        Plug 'junegunn/fzf', {'do': { -> fzf#install() } }
        Plug '/usr/local/opt/fzf'
        set rtp+=~/.fzf
        Plug 'junegunn/fzf.vim'
```

That line above, `set rtp+=~/.fzf` made all the difference.

**Comment by nicknisi** [comment]
This is good info, thanks! I've been neglecting linux because I haven't really had a use for these to run on it personally, but that's changing. I have started looking at it more closely, starting with the install script in #147, starting with a rewrite of the install scripts to be more cross-platform compatible. I'll work on incorporating these fixes as well as the fixes in #136.
